### PR TITLE
Feat: signed sessions

### PR DESCRIPTION
**Pull request by gabrielcipriano**
A proposal for signed endpoints, so we have "authentication" at request level

### using it on client side example:
```ts
import { useCommune } from "@commune-ts/providers/use-commune";
import { signData, createSessionData} from "@commune-ts/api";

const { signHex } = useCommune();

const useStartSession = api.auth.startSession.useMutation();

const handleStartSession = async () => {
  const sessionData = createSessionData(window);
  const signedData = signData(signHex, sessionData);

  const result = await useStartSession.mutateAsync(signedData);

  console.log("token: ", result.token);
  console.log("authenticationType: ", result.authenticationType); // Bearer

  const authorization = `${result.authenticationType} ${result.token}`; // "Bearer <token>"

  // Set the token as authorization header
  // ...
}
```

### CODE REVIEW

File: packages/api/src/auth/signed-data.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return { data: validated.data, address };
+};
+
+export const signData = async <T>(
+  signer: (
+    msgHex: `0x${string}`,
+  ) => Promise<{ signature: `0x${string}`; address: string }>,
+  data: T,
+): Promise<SignedInput> => {
+  const dataHex = stringToHex(JSON.stringify(data));
```
Comment (steinerkelvin) [inline packages/api/src/auth/signed-data.ts:37]: I think we may need JSON canonicalization here, but I'm not sure. Probably not.

Comment (gabrielcipriano) [reply packages/api/src/auth/signed-data.ts:37]: probably not
> Does JSON stringify preserve order?
> Only enumerable own properties are visited. This means Map, Set, etc. will become "{}". You can use the replacer parameter to serialize them to something more useful. Properties are visited using the same algorithm as Object.keys(), which has a well-defined order and is stable across implementations.  **For example, JSON.stringify on the same object will always produce the same string, and JSON.parse(JSON.stringify(obj)) would produce an object with the same key ordering as the original (assuming the object is completely JSON-serializable)**

File: apps/commune-governance/src/app/components/create-comment.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { api } from "~/trpc/react";
 
 const MAX_CHARACTERS = 300;
 const MAX_NAME_CHARACTERS = 300;
-const MIN_STAKE_REQUIRED = 5000;
+const MIN_STAKE_REQUIRED = 10;
```
Comment (vinicius-sacramento) [inline apps/commune-governance/src/app/components/create-comment.tsx:15]: This value needs to be returned to 5000
```suggestion
const MIN_STAKE_REQUIRED = 5000;
```

**Comment by EdSDR** [comment]
This will be closed since front end integration PR has all the contents of this branch

**Pull request closed without merging**